controllers/replication/replication.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package replication
+
+import (
+	"github.com/csi-addons/volume-replication-operator/pkg/client"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// Replication represents the instance of a single replication operation
+type Replication struct {
+	Params      CommonRequestParameters
+	KnownErrors []codes.Code
```

Comment:
this is not needed to be part of this anymore?

Reply:
you mean `knownErrors` need not be part of `Replication` struct?  I think we can still keep it as looks like a legit field for Replication.  Or we can also use `KnownErrors` as variables in each methods like `markVolumeAsPrimary`

Reply:
using `KnownErrors` as a variable.